### ISSUE TITLE
GitHub Desktop reporting "Never Fetched"

### ISSUE THREAD

**Issue by Computerhope** [desktop/desktop#9006]
### Describe the bug

After moving the .git folder out of the working tree GitHub Desktop always reports "Never fetched" even after fetch occurs. Everything still works fine, just always see "Never fetched" message in GitHub Desktop.

### Version & OS

OS: Windows 10
GitHub Desktop: 2.3.1

### Steps to reproduce the behavior

1. Open GitHub Desktop.
2. Open GitHub command line (Ctrl+`).
3. Type a command similar to the following example, depending on where you want to move the .git folder.

git init --separate-git-dir c:\git\example.git

4. Git should give a reinitialized existing Git repository message. Once complete, exit the command line.
5. In GitHub Desktop click the Fetch button. Fetch occurs but now always shows "Never fetched."

### Expected behavior

Should show last Fetch time.

### Actual behavior

Always shows "Never fetched."

### Additional context

I've also tried adding a core.worktree value into the config and get the same problem.

**Comment by steveward** [comment]
Thanks for contributing @Computerhope! Would you mind uploading the log file from GitHub Desktop? I'd like to take a closer look to see what may be happening here. To access the log files go to the file menu in GitHub Desktop and select `Help` > `Show Logs in Finder (macOS) or Explorer (Windows)`.

I would also love to hear about your specific usage of the `--separate-git-dir` feature -- this isn't something that we see being used very often. 

>I've also tried adding a core.worktree value into the config and get the same problem.

We are tracking support for worktrees in https://github.com/desktop/desktop/issues/907 -- you can follow along in that issue for updates about this.

**Comment by Computerhope** [comment]
Hey Steve,

I used the --separate-git-dir to help with compatibility with Dropbox by not syncing the .git directory. I use Dropbox to make immediate backups of files as they're changed and then GitHub to help track changes and work with others.

Attached is the most recent log.

[2020-01-30.desktop.production.log](https://github.com/desktop/desktop/files/4135112/2020-01-30.desktop.production.log)

-Nathan

**Comment by steveward** [comment]
@Computerhope is this still an issue you are seeing in GitHub Desktop? Does it happen in repositories that are not using the `--separate-git-dir` flag?

**Comment by sgbaird** [comment]
I get a similar issue - submodule in GitHub Desktop reports "never fetched", but everything seems to work normally otherwise. I don't use `--separate-git-dir`; at least I've never specified that option.

**Comment by Computerhope** [comment]
@steveward Yes this is still happening. All my repositories are handled this way, but for testing purposes I did clone another existing repository without separating the dir and it does show a last fetched.

The issue seems to only seems to happen on repositories that have a separate git dir, which right now is three different repositories I work in daily. Everything else has been working fine since I made the original post, but it has always said "Never fetched."

Above testing done with latest version: 2.5.3.

**Comment by niik** [comment]
Hi @Computerhope sorry for taking so long to get back to you on this. GitHub Desktop currently assumes that the git dir is always located at `./.git` relative to the repository root. See for example:

https://github.com/desktop/desktop/blob/03c154c463552a338a1eacceb1dd72abfeb42928/app/src/lib/stores/git-store.ts#L1372

As well as

https://github.com/desktop/desktop/blob/03c154c463552a338a1eacceb1dd72abfeb42928/app/src/lib/git/rebase.ts#L98-L117

We'd have to replace all these instances with something derived from `git rev-parse --absolute-git-path` (or similar). Ideally we wouldn't do that all the time but rather cache the result of of `rev-parse` or maybe even persist it alongside the repository path.

I'm going to label this as a priority-3 bug for now given how rare this setup is. As a counter argument this scenario applies to all submodules added directly to the app as well so the next time we do a push to support submodules better we might want to include this.

**Comment by vorou** [comment]
The same thing happens for worktrees, which are probably a little more popular.

(Looks like it should be fixed as a part of this issue, not https://github.com/desktop/desktop/issues/907).

**Comment by arielcarrera** [comment]
same issue after move my directory
my workaround was remove the project from Github Desktop and press "add local repository" again.

**Comment by capr** [comment]
Same problem when trying to open an already-cloned git repo that contains a `worktree` entry in the repo's `.git/config` file. Trying to open the repo results in the following error message saying there's no `x:\luapower` (that being the workdir), but the directory is right there and it's a proper work dir of the repo I'm trying to open.
![gitdesktopbug](https://user-images.githubusercontent.com/131660/143617029-3e609c9e-a116-4a0d-bfd8-c99940c79cd7.png)

**Comment by blawnode** [comment]
@arielcarrera 's solution worked wonders for me. Seems like that's very relevant..! Thanks much, Ariel :)

**Comment by efsalvarenga** [comment]
> @arielcarrera 's solution worked wonders for me. Seems like that's very relevant..! Thanks much, Ariel :)

Did not work for me though. I have the same setup.

**Comment by blawnode** [comment]
As long as the problem persists for some people, there's certainly work to be done.
